### PR TITLE
Fixed #28932 -- Prevented Oracle from truncating trailing zeros in the fractional part of DecimalField.

### PR DESCRIPTION
**Pull request by sir-sigurd**
https://code.djangoproject.com/ticket/28932

### CODE REVIEW

**Comment by sir-sigurd** [comment]
buildbot, test on oracle.

**Comment by felixxm** [comment]
Maybe it will be better to restore `convert_decimalfield_value` on Oracle :thinking:  (removed in https://github.com/django/django/commit/7c1f3901bcdabb1340a621e7df9b24f3acd0d6f3):
```diff
diff --git a/django/db/backends/oracle/operations.py b/django/db/backends/oracle/operations.py
index 6e60719086..0c8d02c46e 100644
--- a/django/db/backends/oracle/operations.py
+++ b/django/db/backends/oracle/operations.py
@@ -157,6 +158,8 @@ END;
                 converters.append(self.convert_datetimefield_value)
         elif internal_type == 'DateField':
             converters.append(self.convert_datefield_value)
+        elif internal_type == 'DecimalField':
+            converters.append(self.convert_decimalfield_value)
         elif internal_type == 'TimeField':
             converters.append(self.convert_timefield_value)
         elif internal_type == 'UUIDField':
@@ -172,6 +175,12 @@ END;
             )
         return converters
 
+    def convert_decimalfield_value(self, value, expression, connection):
+        if value is not None:
+            value = expression.output_field.format_number(value)
+            value = Decimal(value)
+        return value
+
     def convert_textfield_value(self, value, expression, connection):
         if isinstance(value, Database.LOB):
             value = value.read()
```
It also solves the issue.

**Comment by sir-sigurd** [comment]
I think it will be slower however this should be checked.

**Comment by sir-sigurd** [comment]
@felixxm 
**Benchmarks**
in current PR:
```
%timeit for x in DecimalModel.objects.values_list('d'): pass
73 ms ± 1.28 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)
```
with converter in `get_db_converters()`:
```
%timeit for x in DecimalModel.objects.values_list('d'): pass
193 ms ± 10.8 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

File: django/db/backends/oracle/base.py [hunk 1]
```diff
@@ -399,6 +399,14 @@ def __init__(self, connection):
     def _output_number_converter(value):
         return decimal.Decimal(value) if '.' in value else int(value)
 
+    @staticmethod
+    def _get_decimal_converter(precision, scale):
```
Comment (felixxm) [inline django/db/backends/oracle/base.py]: Maybe `_output_decimal_converter` to keep consistency of converters names.

Comment (sir-sigurd) [reply django/db/backends/oracle/base.py]: I tried to emphasize that it's not converter by itself, but returns converter.

**Comment by timgraham** [comment]
Should we limit this change to master and skip the failing test on 2.0?

**Comment by felixxm** [comment]
I think yes.

**Pull request merged**